apps/ecosystem/src/ecosystem.service.ts

```diff
@@ -1557,20 +1663,86 @@ export class EcosystemService {
         );
       }
     } catch (error) {
-      this.logger.error(`In submit transaction: ${JSON.stringify(error)}`);
-      if (error && error?.status && error?.status?.message && error?.status?.message?.error) {
-        throw new RpcException({
-          message: error?.status?.message?.error?.reason
-            ? error?.status?.message?.error?.reason
-            : error?.status?.message?.error,
-          statusCode: error?.status?.code
-        });
-      } else {
-        throw new RpcException(error.response ? error.response : error);
+      this.logger.error(`In submit Indy transaction: ${JSON.stringify(error)}`);
+      this.handleException(error);
+    }
+  }
+
+
+  async submitW3CTransaction(transactionPayload: TransactionPayload): Promise<object> {
+    try {
+      const { endorsementId } = transactionPayload;
+      const endorsementTransactionPayload = await this.ecosystemRepository.getEndorsementTransactionById(
+        endorsementId,
+        endorsementTransactionStatus.REQUESTED
+      );
+
+      if (!endorsementTransactionPayload) {
+        throw new BadRequestException(ResponseMessages.ecosystem.error.transactionNotRequested);
       }
+
+      const w3cEndorsementTransactionPayload = await this.ecosystemRepository.getEndorsementTransactionByIdAndType(endorsementId, endorsementTransactionType.W3CSCHEMA);
```

Comment:
it's better to destructure and use the response

Reply:
Resolved